docs/docs/lists-conditional-rendering.md

```diff
@@ -0,0 +1,264 @@
+
+```javascript
+function Number (props) {
+  return <li>{props.value}</li>  
+}
+function Numbers(props) {
+  const numbers = [1, 2, 3, 4, 5];
+  return (
+    <div>
+      <ul>
+        {numbers.map((number, index) => <Number key={index} value={number} />)}
+      </ul>
+    </div>
+  );
+}
+```
+
+<a target="_blank" href="https://codepen.io/ericnakagawa/pen/gwjdzN?editors=0011">Try it out on JSFiddle.</a>
+
+
+### Keys
+
+In the previous example we provide the index of an item in our array as our `key`. It is important to always provide keys to your lists of elements.
+
+* For lists that are unlikely or won't change, using an item index is fine.
+* For lists that may change, it is important to use a unique key. A string representing identity of the data item can also work.
+
+* Keys should be unique amongst siblings. This means that the keys used with groups of related elements in a list should be unique .
+* You will be warned when you don't include keys.
+
+#### Performance
+
+In React, the absence of keys on an element can impact your application performance.
+
+When a `key` is provided, React chooses the most performant method for updating the rendered objects.
+
+#### Example 1
+
+Given the HTML below:
+
+```html
+<ul>
+  <li>1</li> /* key: "1" */
+  <li>2</li> /* key: "2" */
+</ul>
+```
+
+In the event that the element containing "2" needs to be removed, React will hide the element instead of removing the object entirely like so:
+
+```html
+<ul>
+  <li>1</li> /* key: "1" */
+  <li style={display: 'none'}>2</li> /* key: "2" */
+</ul>
+```
+
+#### Example 2
+
+Given the html below:
+
+```html
+<ul>
+  <li>1</li> /* key: "1" */
+  <li>2</li> /* key: "2" */
+</ul>
+```
+
+In the event that the element "1" needs to be removed, React will swap the text and hide the second element like so:
+
+```html
+<ul>
+  <li>2</li> /* key: "2" */
+</ul>
+```
+
+## Conditional Rendering
+
+With React you can render distinct components depending on the value of your data. This means no longer needing to manually modify the CSS classes of HTML elements. Instead, you create distinct components that encapsulate behavior you need. Then, render only the specific component you need depending on the state of your application.
+
+This follows the concept of "separation of concerns" by allowing each component to manage how it should be rendered.
+
+### Element Variables
+
+Given the current state of your application you can assign elements as the value of variables, then only render the needed element.
+
+In the example below we determine which button to display to a potential user. We track state using `this.state.loggedIn` to simulate whether or not they are logged in. If the user is logged in then we want to render a `<LogoutButton />`, otherwise we want `<LoginButton />`. We use a variable `loginButton` to hold either element depending on the value of `this.state.loggedIn` until we render.
```

Comment:
Could you make it into two sentences instead of the `, otherwise` ? E.g.

"In the example below we determine which button to display to a potential user. We track state using `this.state.loggedIn` to represent whether or not they are logged in. If the user is logged in then we want to render a `<LogoutButton />`, otherwise we want a `<LoginButton />`. We use a variable `loginButton` to hold either element, depending on the value of `this.state.loggedIn`."